disqus/disqus.php

```diff
@@ -511,7 +511,13 @@ function dsq_request_handler() {
                     }
                 } else {
                     $ts = time() + 300;
-                    wp_schedule_single_event($ts, 'dsq_sync_forum');
+                    $next_scheduled = wp_next_scheduled('dsq_sync_forum');
+                    if($next_scheduled) {
+                        // error_log("Not scheduling dsq_sync_forum because it's already scheduled for " . $next_scheduled);
+                    } else {
+                        // error_log("Scheduling for $ts because dsq_sync_forum is not currently scheduled.");
+                        wp_schedule_single_event($ts, 'dsq_sync_forum');
+                    }
```

Comment:
I would like to mention my observation. Above if else check is also costly operation. I wouldn't recommend enabling it. 

Reason: On Crunchify we have seen CPU issue for long time and it's very long story with Disqus. I tried numerous experiments and it seems above fix cause higher CPU usage increase from ~1% to ~15% consistent. After a week experiment i remove above code block yesterday and finally CPU usage back to ~1%. NOTE: I do have dedicated server hosting.

<img width="466" alt="disqus cpu usage" src="https://cloud.githubusercontent.com/assets/8164601/17705808/573a619a-639f-11e6-9b29-67516753eb17.png">